src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -487,51 +487,216 @@ void checkConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNode syn
                 }
                 else if (conversion.IsSpan)
                 {
+                    Debug.Assert(source.Type is not null);
                     Debug.Assert(destination.OriginalDefinition.IsSpan() || destination.OriginalDefinition.IsReadOnlySpan());
 
                     CheckFeatureAvailability(syntax, MessageID.IDS_FeatureFirstClassSpan, diagnostics);
 
-                    // PROTOTYPE: Check runtime APIs used for other span conversions once they are implemented.
                     // NOTE: We cannot use well-known members because per the spec
                     // the Span types involved in the Span conversions can be any that match the type name.
-                    if (TryFindImplicitOperatorFromArray(destination.OriginalDefinition) is { } method)
+
+                    // Span<T>.op_Implicit(T[]) or ReadOnlySpan<T>.op_Implicit(T[])
+                    if (source.Type is ArrayTypeSymbol)
                     {
-                        diagnostics.ReportUseSite(method, syntax);
+                        reportUseSiteOrMissing(
+                            TryFindImplicitOperatorFromArray(destination.OriginalDefinition),
+                            destination.OriginalDefinition,
+                            WellKnownMemberNames.ImplicitConversionName,
+                            syntax,
+                            diagnostics);
                     }
-                    else
+
+                    // ReadOnlySpan<T> Span<T>.op_Implicit(Span<T>)
+                    if (source.Type.OriginalDefinition.IsSpan())
```

Comment:
>OriginalDefinition.IsSpan() [](http://example.com/codeflow?start=36&length=27)

Is `.OriginalDefinition` necessary for the calls to `IsSpan()` and `IsReadOnlySpan()`? If not, consider removing `.OriginalDefinition` in those cases, since there are many uses of that here.